l5kit/l5kit/environment/reward.py

```diff
@@ -0,0 +1,119 @@
+from l5kit.simulation.unroll import SimulationOutputCLE
+
+
+class Reward(ABC):
+    """Base class interface for gym environment reward."""
+    #: The prefix that will identify this reward class
+    reward_prefix: str
+
+    @abstractmethod
+    def reset(self) -> None:
+        """Reset the reward state when new episode starts.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_reward(self, frame_index: int, simulated_outputs: List[SimulationOutputCLE]) -> Dict[str, float]:
+        """Return the reward at a particular time-step during the episode.
+
+        :param frame_index: the frame index for which reward is to be calculated
+        :param simulated_outputs: the object contain the ego target and prediction attributes
+        :return: reward at a particular frame index (time-step) during the episode containing total reward
+            and individual components that make up the reward.
+        """
+        raise NotImplementedError
+
+
+class L2DisplacementYawReward(Reward):
+    """This class is responsible for calculating a reward based on
+    (1) L2 displacement error on the (x, y) coordinates
+    (2) Closest angle error on the yaw coordinate
+    during close loop simulation within the gym-compatible L5Kit environment.
+
+    :param reward_prefix: the prefix that will identify this reward class
+    :param metric_set: the set of metrics to compute
+    :param enable_clip: flag to determine whether to clip reward
+    :param rew_clip_thresh: the threshold to clip the reward
+    :param use_yaw: flag to penalize the yaw prediction
+    :param yaw_weight: weight of the yaw error
+    """
+
+    def __init__(self, reward_prefix: str = "L2DisplacementYaw", metric_set: Optional[L5MetricSet] = None,
+                 enable_clip: bool = True, rew_clip_thresh: float = 15.0,
+                 use_yaw: Optional[bool] = True, yaw_weight: Optional[float] = 20.0) -> None:
+        """Constructor method
+        """
+        self.reward_prefix = reward_prefix
+        # Metric Set
+        self.metric_set = metric_set if metric_set is not None else L2DisplacementYawMetricSet()
+
+        # Verify that error metrics necessary for reward calculation are present in the metric set
+        metric_names = [metric.metric_name for metric in self.metric_set.evaluation_plan.metrics]
+        if 'yaw_error_closest_angle' not in metric_names:
+            raise RuntimeError('\'yaw_error_closest_angle\' missing in metric set')
+        if 'displacement_error_l2' not in metric_names:
+            raise RuntimeError('\'displacement_error_l2\' missing in metric set')
+
+        self.use_yaw = use_yaw
+        self.yaw_weight = yaw_weight
+
+        self.enable_clip = enable_clip
+        self.rew_clip_thresh = rew_clip_thresh
+
+    def reset(self) -> None:
+        """Reset the closed loop evaluator when a new episode starts.
+        """
+        self.metric_set.reset()
+
+    def slice_simulated_output(self, index: int,
```

Comment:
This looks like a static method, can you add the `staticmethod` decorator and remove the self ? As it is not using  `self` for anything.